lightning/src/events/mod.rs

```diff
@@ -1766,11 +1799,28 @@ impl Writeable for Event {
 					(8, path.blinded_tail, option),
 				})
 			},
-			&Event::HTLCHandlingFailed { ref prev_channel_id, ref failed_next_destination } => {
+			&Event::HTLCHandlingFailed { ref prev_channel_id, ref failure_type, ref failure_reason } => {
 				25u8.write(writer)?;
+
+				// The [`HTLCHandlingType::UnknownNextPeer`] variant is deprecated, but we want to
+				// continue writing it to allow downgrading. Detect the case where we're
+				// representing it as [`HTLCHandlingType::InvalidForward`] and
+				// [`LocalHTLCFailureReason::UnknownNextHop`] and write the old variant instead.
+				let downgradable_type = match (failure_type, failure_reason) {
+					(HTLCHandlingFailureType::InvalidForward { requested_forward_scid },
+						Some(HTLCHandlingFailureReason::Local {
+							reason: LocalHTLCFailureReason::UnknownNextPeer
+						}))
+						=> HTLCHandlingFailureType::UnknownNextHop {
+							requested_forward_scid: *requested_forward_scid,
+						},
+					_ => failure_type.clone()
+				};
```

Comment:
Apologies if I signed off on this previously, but I think it would make more sense to always write the non-deprecated type and then include a release note that downgrades will result in reading the deprecated type as `::InvalidForward`, since it'll save some complexity here and keep the door open for us to delete the variant in the future.

Reply:
Having done it this way, I think it looks much better 👍  doesn't seem like _that_ much of a loss to have less information on downgrade in comparison

Reply:
Are there no functional consequences after the downgrade for this loss of information?

Reply:
afaik no, `HTLCHandlingFailed` is a "read-only" event - it doesn't require any action from an end user.

So worst case it's a few wrong entries in a tracking dashboard, but I think the release note is sufficient to highlight that (so that downgraders are aware).